### PR TITLE
Use Gradle version catalog to simplify dependencies

### PR DESCRIPTION
**Pull request by ghostbear**
Replaces manual dependencies declarations with type-safe and centralized ones generated by Gradle version catalog.

### Usage
**Versions**
If an extension needs a defined version they can use 
```kt
val version = libs.versions.kotlin.version.get()
```

**Libraries**
If an extension needs a defined library they can use 
```kt
compileOnly(libs.tachiyomi.lib)
```

**Bundles**
If an extension needs a defined bundle they can use 
```kt
compileOnly(libs.bundles.coroutines)
```

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -130,12 +130,12 @@ For example, an extension that needs coroutines, it could add the following:
 
 ```gradle
 dependencies {
-    compileOnly 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.5.2'
-    compileOnly 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.5.2'
+    compileOnly libs.bundles.coroutines
```
Comment (CarlosEsco) [inline CONTRIBUTING.md:133]: can you fix this.   I realize looking this is groovy gradle example

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+rxandroid = { module = "io.reactivex:rxandroid", version = "1.2.1" }
+rxjava = { module = "io.reactivex:rxjava", version = "1.3.8" }
+
+kotlinx-coroutines-core = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core", version.ref = "coroutines_version" }
+kotlinx-coroutines-android = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-android", version.ref = "coroutines_version" }
+
+[bundles]
+common = ["tachiyomi-lib", "jsoup", "okhttp", "kotlin-stdlib", "injekt-core", "rxjava", "kotlin-protobuf", "kotlin-json"]
+coroutines = ["kotlinx-coroutines-core", "kotlinx-coroutines-android"]
+reactivex = ["rxandroid", "rxjava"]
```
Comment (arkon) [inline gradle/libs.versions.toml:29]: Does it make sense to include `rxjava` if it's already in `common`, which every extension should already use?

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[versions]
+kotlin_version = "1.4.32"
+coroutines_version = "1.4.3"
+
+[libraries]
```
Comment (arkon) [inline gradle/libs.versions.toml:5]: Not too familiar with the version catalog stuff, but would it make sense to group the plugins in its own section?

File: gradle/libs.versions.toml (hunk 2 above)
Comment (ghostbear) [reply gradle/libs.versions.toml:29]: In term of `reactivex` as a bundle, yes. In term of usage with other bundles, no. Will change it only have `rxandroid` in the bundle and keep the bundle if anything ever changes

**Comment by arkon** [comment]
This somehow broke when running the build workflow (https://github.com/tachiyomiorg/tachiyomi-extensions/actions/runs/1906414440), so will need to revisit this.

**Comment by arkon** [comment]
Actually, it may be unrelated... https://github.com/tachiyomiorg/tachiyomi-extensions/runs/5350875654?check_suite_focus=true

**Pull request merged**